src/components/BookPlace.tsx

```diff
@@ -1,53 +1,175 @@
 
 interface BookPlaceProps {
   placeId: string
   handleClose: () => void
   placeBookings: BookingPlaceData[]
-  placeName: string 
+  placeName: string
 }
 
 const BookPlace: React.FC<BookPlaceProps> = ({ handleClose, placeId, placeBookings, placeName }) => {
+  // Getting info from inputs
+  const [visitorName, setVisitorName] = useState<string>('')
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => setVisitorName(event.target.value)
+
+  const [amount, setAmount] = useState<number>()
+
+  const [isFormsDisabled, setIsFormsDisabled] = useState(false)
+
+  const sendPayment = (amount: number, paidDays: FiniteDateRange) => {
+    setAmount(amount)
+    setPaidDays(paidDays)
+  }
+
+  // Getting and sending data with firebase
   const { book } = useBookingPlace(placeId)
+  const bookPlace = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    setIsFormsDisabled(true)
 
-  const [dateRange, setDateRange] = useState<DateRange>([null, null])
-  const handleDateRangeChange = (dateRange: DateRange) => setDateRange(dateRange)
+    if (bookedDays && paidDays && visitorName && amount) {
+      book(
+        {
+          startDate: Timestamp.fromDate(bookedDays.startDate),
+          endDate: bookedDays.endDate ? Timestamp.fromDate(bookedDays.endDate) : null,
+          visitorName,
+          placeName,
+          amount,
+          paidDays: {
+            startDate: Timestamp.fromDate(paidDays.startDate),
+            endDate: Timestamp.fromDate(paidDays.endDate)
+          }
+        }, placeBookings
+      )
+      .then(() => {
+        AppToaster.show({ message: 'The place has been booked' })
+        setIsFormsDisabled(false)
+        handleClose()
+      })
+      .catch(({ message }) => {
+        AppToaster.show({ message, intent: 'danger' })
+        setIsFormsDisabled(false)
+      })
+    }
+  }
 
-  const [visitorName, setVisitorName] = useState('')
-  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => setVisitorName(event.target.value)
+  // Managing modal windows
+  const [isBookingDatesOpen, setIsBookingDatesOpen] = useState(false)
+  const handleBookingDateOpen = () => setIsBookingDatesOpen(true)
+  const handleBookingDatesClose = () => setIsBookingDatesOpen(false)
+
+  const [isPaymentOpen, setIsPaymentOpen] = useState(false)
+  const handlePaymentOpen = () => setIsPaymentOpen(true)
+  const handlePaymentClose = () => setIsPaymentOpen(false)
 
-  const [amount, setAmount] = useState('')
-  const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => setAmount(event.target.value)
+  // Working with dates
+  const [bookedDays, setBookedDays] = useState<CustomDateRange>(null)
+  const [paidDays, setPaidDays] = useState<FiniteDateRange | null>(null)
+  const currentDay = new Date()
+
+  const [foreverBooking, setForeverBooking] = useState(false)
+  const handleForeverBooking = () => {
+    setForeverBooking(!foreverBooking)
+    setBookedDays(null)
```

Comment:
You should not reset the selected dates.
Just don't show and don't save the end date if foreverFlag is enabled.